contracts/GenArt721CoreV3_Engine.sol

```diff
@@ -0,0 +1,2137 @@
+ * - proposeArtistPaymentAddressesAndSplits (Note that this has to be accepted
+ *   by adminAcceptArtistAddressesAndSplits to take effect, which is restricted
+ *   to the Admin ACL contract, or the artist if the core contract owner has
+ *   renounced ownership. Also note that a proposal will be automatically
+ *   accepted if the artist only proposes changed payee percentages without
+ *   modifying any payee addresses, or is only removing payee addresses.)
+ * - toggleProjectIsPaused (note the artist can still mint while paused)
+ * - updateProjectSecondaryMarketRoyaltyPercentage (up to
+     ARTIST_MAX_SECONDARY_ROYALTY_PERCENTAGE percent)
+ * - updateProjectWebsite
+ * - updateProjectMaxInvocations (to a number greater than or equal to the
+ *   current number of invocations, and less than current project maximum
+ *   invocations)
+ * - updateProjectBaseURI (controlling the base URI for tokens in the project)
+ * ----------------------------------------------------------------------------
+ * The following function is restricted to either the Admin ACL contract, or
+ * the Artist address if the core contract owner has renounced ownership:
+ * - adminAcceptArtistAddressesAndSplits
+ * - updateProjectArtistAddress (owner ultimately controlling the project and
+ *   its and-on revenue, unless owner has renounced ownership)
+ * ----------------------------------------------------------------------------
+ * The following function is restricted to the artist when a project is
+ * unlocked, and only callable by Admin ACL contract when a project is locked:
+ * - updateProjectDescription
+ * ----------------------------------------------------------------------------
+ * The following function is restricted to owner calling directly:
+ * - transferOwnership
+ * - renounceOwnership
+ * ----------------------------------------------------------------------------
+ * The following configuration variables are set at time of contract deployment,
+ * and not modifiable thereafter (immutable after the point of deployment):
+ * - (bool) autoApproveArtistSplitProposals
+ * ----------------------------------------------------------------------------
+ * Additional admin and artist privileged roles may be described on minters,
+ * registries, and other contracts that may interact with this core contract.
+ */
+contract GenArt721CoreV3_Engine is
+    ERC721_PackedHashSeed,
+    Ownable,
+    IGenArt721CoreContractV3_Engine
+{
+    using BytecodeStorage for string;
+    using BytecodeStorage for address;
+    using Bytes32Strings for bytes32;
+    using Strings for uint256;
+    uint256 constant ONE_HUNDRED = 100;
+    uint256 constant ONE_MILLION = 1_000_000;
+    uint24 constant ONE_MILLION_UINT24 = 1_000_000;
+    uint256 constant FOUR_WEEKS_IN_SECONDS = 2_419_200;
+    uint8 constant AT_CHARACTER_CODE = uint8(bytes1("@")); // 0x40
+
+    // numeric constants
+    uint256 constant MAX_SECONDARY_SALES_BPS = 10000; // 10_000 BPS = 100%
```

Comment:
WDYT about naming this `MAX_PROVIDER_SECONDARY_SALES_BPS` (since it is the max of sum render provider and platform provider secondary sales bps values)

Reply:
I dig it ser–updating!